feat: constraining slots (#7758)
AztecProtocol/aztec-packages: docs/docs/protocol-specs/l1-smart-contracts/index.md

The contract keeps track of the current state of the L2 and progresses this state when a valid L2 block is received.
It also facilitates cross-chain communication (communication between the L1 inbox and outbox contracts).

:::info

> The following is simplified by showcasing a case where proof and block is provided at the same time.

Reviewer: ```suggestion
The following example shows a simplified case where proof and block are provided in the same transaction.
```
